ql/src/java/org/apache/hadoop/hive/ql/session/SessionState.java

```diff
@@ -388,6 +397,19 @@ public boolean getIsQtestLogging() {
     return isQtestLogging;
   }
 
+  public PathCleaner getPathCleaner() {
+    if (pathCleaner == null) {
+      pathCleaner = new PathCleaner(getSessionId());
+      pathCleaner.start();
+    }
+    return pathCleaner;
+  }
+
+  public void setSyncCleanup() {
+    pathCleaner = null;
```

Comment:
I think in case `pathCleaner` is initialized - after calling this a thread will be left running in the background